src/Main.hs

```diff
@@ -25,33 +25,36 @@ data Command
 
 main :: IO ()
 main = do
-  -- URI of the postgrest service
-  uri <- mkURI . T.pack . fromMaybe "http://localhost:3000" =<< lookupEnv "TODO_URI"
+  -- Connection to the postgrest service
+  conn <- maybe
+    (TR.TCP <$> (mkURI . T.pack . fromMaybe "http://localhost:3000" =<< lookupEnv "TODO_URI"))
+    (pure . TR.UnixSocket)
+    =<< lookupEnv "PGRST_SERVER_UNIX_SOCKET"
```

Comment:
Prefer clarity over magic/conciseness. It looks like this is equivalent to:

```haskell
getConnection :: IO TR.Connection
getConnection = do
  unixSocket <- lookupEnv "PGRST_SERVER_UNIX_SOCKET"
  case unixSocket of
    Just socket -> return $ TR.UnixSocket socket
    Nothing -> do
      todoUri <- lookupEnv "TODO_URI"
      let defaultUri = "http://localhost:3000"
      let uri = fromMaybe defaultUri todoUri
      tcpConnection <- TR.TCP <$> (mkURI $ T.pack uri)
      return tcpConnection
```

So the user must set `PGRST_SERVER_UNIX_SOCKET` *or* `TODO_URI`?

From UX point of view, this is not optimal. You should overload TODO_URI to support unix domain sockets, for e.g.: `TODO_URI=./data/foo.socket` or `TODO_URI=unix:./data/foo.socket`

Also, what should change in README.md for this change?

Reply:
I was planning to stick to ENVs used by `postgrest` itself, rather than making up my own, like `TODO_URI`. `TODO_URI` in a different PR will get replaced by `PGRST_SERVER_HOST:PGRST_SERVER_PORT` (see https://docs.postgrest.org/en/stable/references/configuration.html#server-host).

Since that would be out of scope for this PR, I didn’t do that here.

Reply:
> `TODO_URI` in a different PR will get replaced by `PGRST_SERVER_HOST:PGRST_SERVER_PORT`

What are the env vars, and their values, when using unix socket?

Reply:
`PGRST_SERVER_UNIX_SOCKET` and `PGRST_SERVER_UNIX_SOCKET_MODE`

https://docs.postgrest.org/en/stable/references/configuration.html#server-unix-socket

Reply:
> Prefer clarity over magic/conciseness. It looks like this is equivalent to:

Thanks! applied in https://github.com/juspay/todo-app/pull/20/commits/35ace49ad94b8312b0f4d84804c10e8d9351dfed

Reply:
> I was planning to stick to ENVs used by postgrest itself, rather than making up my own, like TODO_URI. TODO_URI in a different PR will get replaced by PGRST_SERVER_HOST:PGRST_SERVER_PORT

Done in https://github.com/juspay/todo-app/pull/21